presets/ragengine/models.py

```diff
@@ -25,6 +25,13 @@ class IndexRequest(BaseModel):
     index_name: str
     documents: List[Document]
 
+class UpdateDocumentRequest(BaseModel):
+    doc_id: str
+    text: str
+    hash_value: Optional[str] = None
+    metadata: Optional[dict] = Field(default_factory=dict)
+    is_truncated: bool = False
```

Comment:
I'm thinking we have doc_id, text, etc. in the document class, what do you think? 

Then its 
class UpdateRequest(BaseModel):
    documents: List[Document]
    ...maybe couple other fields need be....

Reply:
Love it! Will make the change